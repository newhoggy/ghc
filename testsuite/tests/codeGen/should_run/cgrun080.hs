{-# OPTIONS_GHC -msse4 #-}
{-# OPTIONS_GHC -mavx #-}
{-# LANGUAGE MagicHash #-}
{-# LANGUAGE UnboxedTuples #-}
-- !!! test the packing of four floats into a vector

import GHC.Exts

data FloatX4 = FX4# FloatX4#

instance Show FloatX4 where
  show (FX4# f) = case (unpackFloatX4# f) of
    (# a, b, c, d #) -> show ((F# a), (F# b), (F# c), (F# d))


main :: IO ()
main = do
  print (FX4# (packFloatX4# (# 9.2#, 8.15#, 7.0#, 6.4# #)))
