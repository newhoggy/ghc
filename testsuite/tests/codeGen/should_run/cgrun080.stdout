(9.2,8.15,7.0,6.4)
